### PR TITLE
Update cosign verify instructions for 2.0

### LINKED ISSUE #39775: Update page Verify Signed Kubernetes Artifacts for cosign 2.0
**This is a Feature Request**
After [cosign 2.0](https://blog.sigstore.dev/cosign-2-0-released/) release, we don't need `COSIGN_EXPERIMENTAL=1`. We should update the instructions on the task [page](https://kubernetes.io/docs/tasks/administer-cluster/verify-signed-artifacts/) to include verification using cosign 2.0

**What would you like to be added**
<details>
<summary> verification using cosign 2.0</summary>

```bash
~ $ cosign verify registry.k8s.io/kube-apiserver:v1.26.0 --certificate-identity-regexp krel-trust@k8s-releng-prod.iam.gserviceaccount.com --certificate-oidc-issuer https://accounts.google.com | jq .

Verification for registry.k8s.io/kube-apiserver:v1.26.0 --
The following checks were performed on each of these signatures:
  - The cosign claims were validated
  - Existence of the claims in the transparency log was verified offline
  - The code-signing certificate was verified using trusted certificate authority certificates
[
  {
    "critical": {
      "identity": {
        "docker-reference": "europe-west1-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/kube-apiserver"
      },
      "image": {
        "docker-manifest-digest": "sha256:d230a0b88a3daf14e4cce03b906b992c8153f37da878677f434b1af8c4e8cc75"
      },
      "type": "cosign container image signature"
    },
    "optional": {
      "1.3.6.1.4.1.57264.1.1": "https://accounts.google.com",
      "Bundle": {
        "SignedEntryTimestamp": "MEYCIQDnvT+cu60FogfZIdroUCP3Mrckp9aILhcvFik9k+NoLwIhAMoZ+shE4tj4SNbN+a29YfAmloCm3S8tZ3jCDbC4o+/B",
        "Payload": {
          "body": "eyJhcGlWZXJzaW9uIjoiMC4wLjEiLCJraW5kIjoiaGFzaGVkcmVrb3JkIiwic3BlYyI6eyJkYXRhIjp7Imhhc2giOnsiYWxnb3JpdGhtIjoic2hhMjU2IiwidmFsdWUiOiI5NGYyZTZlY2FiNTY3YzcwMjQxYjNhN2M5ZTNiMzZlYjg1ZGFkOTEzZDY5M2M5MTgzNmY3MGJkNTNhZjljODI5In19LCJzaWduYXR1cmUiOnsiY29udGVudCI6Ik1FVUNJUUM2b1BTbSs1cldHY2ptTnR0eTR6dHVUeVBMRnM5dlk5eFZVUmswaWZjWENRSWdRdURQOGlxbzB2QysweGxHTm11QU5VUGx1QVoxZEpSd0FGRzlxdXY2RlY4PSIsInB1YmxpY0tleSI6eyJjb250ZW50IjoiTFMwdExTMUNSVWRKVGlCRFJWSlVTVVpKUTBGVVJTMHRMUzB0Q2sxSlNVTjJla05EUVd0VFowRjNTVUpCWjBsVlIxSkZhVXhqVDJjdlYySlFNekkwUlhodFRsTk9Xamt6TlhaemQwTm5XVWxMYjFwSmVtb3dSVUYzVFhjS1RucEZWazFDVFVkQk1WVkZRMmhOVFdNeWJHNWpNMUoyWTIxVmRWcEhWakpOVWpSM1NFRlpSRlpSVVVSRmVGWjZZVmRrZW1SSE9YbGFVekZ3WW01U2JBcGpiVEZzV2tkc2FHUkhWWGRJYUdOT1RXcEplRTFxUVRSTmFrMTVUMFJSZVZkb1kwNU5ha2w0VFdwQk5FMXFUWHBQUkZGNVYycEJRVTFHYTNkRmQxbElDa3R2V2tsNmFqQkRRVkZaU1V0dldrbDZhakJFUVZGalJGRm5RVVV6VEZGT1VsSkdXREJPY1UwMFUyRlBlVWhuWWxCNmVrWXdTVEF5VkRWUldrTkpNMEVLZVdkaGNIaFRlVU5JYUVWa1F5OVljV3BxZEU5MlZXdFFRamRYUjNwYVF6QmpaMUZOZFhaM1pIWnZURFpXTVVoNFNYRlBRMEZYVFhkblowWm1UVUUwUndwQk1WVmtSSGRGUWk5M1VVVkJkMGxJWjBSQlZFSm5UbFpJVTFWRlJFUkJTMEpuWjNKQ1owVkdRbEZqUkVGNlFXUkNaMDVXU0ZFMFJVWm5VVlZpYUU5NENrMW1WRlJxZEVwMk4wdHNaRmRUUkVoNFJYaG1TVUZKZDBoM1dVUldVakJxUWtKbmQwWnZRVlV6T1ZCd2VqRlphMFZhWWpWeFRtcHdTMFpYYVhocE5Ga0tXa1E0ZDFGQldVUldVakJTUVZGSUwwSkVXWGRPU1VWNVlUTktiR0pETVRCamJsWjZaRVZDY2s5SVRYUmpiVlp6V2xjMWJreFlRbmxpTWxGMVlWZEdkQXBNYldSNldsaEtNbUZYVG14WlYwNXFZak5XZFdSRE5XcGlNakIzUzFGWlMwdDNXVUpDUVVkRWRucEJRa0ZSVVdKaFNGSXdZMGhOTmt4NU9XaFpNazUyQ21SWE5UQmplVFZ1WWpJNWJtSkhWWFZaTWpsMFRVbEhTMEpuYjNKQ1owVkZRV1JhTlVGblVVTkNTSGRGWldkQ05FRklXVUV6VkRCM1lYTmlTRVZVU21vS1IxSTBZMjFYWXpOQmNVcExXSEpxWlZCTE15OW9OSEI1WjBNNGNEZHZORUZCUVVkRk9VSklabFJuUVVGQ1FVMUJVbnBDUmtGcFJVRjVhRlZKUzNkUVJRcHVaMEZTYlZGeU5tMTZlazlpZERkalpESXhhMjV2TjFSaGEycDVVMUlyYXpGRk1FTkpRMlp0TDJ4elQwaDNjM05sVWpod1oyUm9VRWRHZW1GaVZVTjZDakpNUXpBNVRuUmpVM1JSYlRKWWFXVk5RVzlIUTBOeFIxTk5ORGxDUVUxRVFUSnJRVTFIV1VOTlVVTlJkRWRQZWs1YUswWldkU3RpUXk4d1FXVm9SSFlLUmtaclMya3liMkZNVldkMWVHeDVaMHN2YmtKUVZtcE9aakkxZVVoWlFXazNXVTF2Ym01UFQzRXdORU5OVVVRck5XVnVibEJpVUZWNGVrTkVhMGRqZEFwT1YyNTFUVEpSZVZVclVXUmhSRkJEWW1oYWFHTkxUelZRWjNSRVJUaDRWRkZ5TTBRNVprVTFPVU4yWmpFNWJ6MEtMUzB0TFMxRlRrUWdRMFZTVkVsR1NVTkJWRVV0TFMwdExRbz0ifX19fQ==",
          "integratedTime": 1670542146,
          "logIndex": 8691003,
          "logID": "c0d23d6ad406973f9559f3ba2d1ca01f84147d8ffc5b8445c224f98b9591801d"
        }
      },
      "Issuer": "https://accounts.google.com",
      "Subject": "krel-trust@k8s-releng-prod.iam.gserviceaccount.com",
      "org.kubernetes.kpromo.mirrors": "asia-northeast2-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/kube-controller-manager-amd64,us-west1-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/kube-controller-manager-amd64,us.gcr.io/k8s-artifacts-prod/kubernetes/kube-controller-manager-amd64,us-south1-docker.pkg.dev/k8s-artifacts-prod/images/kube-controller-manager-amd64,us-west2-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/kube-controller-manager-amd64,europe-west9-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/kube-controller-manager-amd64,us-west1-docker.pkg.dev/k8s-artifacts-prod/images/kube-controller-manager-amd64,asia-northeast1-docker.pkg.dev/k8s-artifacts-prod/images/kube-controller-manager-amd64,europe-north1-docker.pkg.dev/k8s-artifacts-prod/images/kube-controller-manager-amd64,europe-southwest1-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/kube-controller-manager-amd64,asia-east1-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/kube-controller-manager-amd64,australia-southeast1-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/kube-controller-manager-amd64,us-east4-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/kube-controller-manager-amd64,asia-northeast2-docker.pkg.dev/k8s-artifacts-prod/images/kube-controller-manager-amd64,us-south1-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/kube-controller-manager-amd64,europe-west1-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/kube-controller-manager-amd64,asia.gcr.io/k8s-artifacts-prod/kubernetes/kube-controller-manager-amd64,us-central1-docker.pkg.dev/k8s-artifacts-prod/images/kube-controller-manager-amd64,asia-northeast1-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/kube-controller-manager-amd64,us-east1-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/kube-controller-manager-amd64,asia.gcr.io/k8s-artifacts-prod/kube-controller-manager-amd64,europe-west2-docker.pkg.dev/k8s-artifacts-prod/images/kube-controller-manager-amd64,southamerica-west1-docker.pkg.dev/k8s-artifacts-prod/images/kube-controller-manager-amd64,europe-west8-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/kube-controller-manager-amd64,us-east5-docker.pkg.dev/k8s-artifacts-prod/images/kube-controller-manager-amd64,asia-south1-docker.pkg.dev/k8s-artifacts-prod/images/kube-controller-manager-amd64,us-east4-docker.pkg.dev/k8s-artifacts-prod/images/kube-controller-manager-amd64,us-west2-docker.pkg.dev/k8s-artifacts-prod/images/kube-controller-manager-amd64,eu.gcr.io/k8s-artifacts-prod/kube-controller-manager-amd64,europe-north1-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/kube-controller-manager-amd64,southamerica-west1-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/kube-controller-manager-amd64,europe-west4-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/kube-controller-manager-amd64,eu.gcr.io/k8s-artifacts-prod/kubernetes/kube-controller-manager-amd64,us.gcr.io/k8s-artifacts-prod/kube-controller-manager-amd64,europe-west8-docker.pkg.dev/k8s-artifacts-prod/images/kube-controller-manager-amd64,asia-south1-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/kube-controller-manager-amd64,europe-west9-docker.pkg.dev/k8s-artifacts-prod/images/kube-controller-manager-amd64,australia-southeast1-docker.pkg.dev/k8s-artifacts-prod/images/kube-controller-manager-amd64,us-central1-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/kube-controller-manager-amd64,asia-east1-docker.pkg.dev/k8s-artifacts-prod/images/kube-controller-manager-amd64,us-east5-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/kube-controller-manager-amd64,europe-west2-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/kube-controller-manager-amd64,europe-southwest1-docker.pkg.dev/k8s-artifacts-prod/images/kube-controller-manager-amd64,europe-west4-docker.pkg.dev/k8s-artifacts-prod/images/kube-controller-manager-amd64,us-east1-docker.pkg.dev/k8s-artifacts-prod/images/kube-controller-manager-amd64,europe-west1-docker.pkg.dev/k8s-artifacts-prod/images/kube-controller-manager-amd64"
    }
  }
]
```
Add a note if cosign version < 2.0
</details>

**Why is this needed**
There are some breaking changes in cosign 2.0 and we should update the instruction for the verification of images. 
**Comments**

### PR DESCRIPTION
**Pull request by mrgiles**
Cosign 2.0 introduced significant command syntax changes to verify images and blobs. Fixes #39775

### CODE REVIEW

**Comment by mrgiles** [comment]
The following values for the `cosign` command flags have been verified by @saschagrunert, as requested by @PushkarJ  

For `--certificate-oidc-issuer`:
- `https://accounts.google.com`

For `--certificate-identity` :
- `krel-trust@k8s-releng-prod.iam.gserviceaccount.com` (verify)
- `krel-staging@k8s-releng-prod.iam.gserviceaccount.com` (verify-blob)

The 'commands to verify images (`cosign verify`) and blobs (`cosign verify-blob`) listed in the solutions run successfully in my tests using Cosign 2.0.

FYI @sftim

**Comment by mrgiles** [comment]
**`cosign verify-blob` test:**

```bash
URL=https://dl.k8s.io/release/v1.27.1/bin/linux/amd64
BINARY=kubectl
FILES=(
    "$BINARY"
    "$BINARY.sig"
    "$BINARY.cert"
)
for FILE in "${FILES[@]}"; do
    curl -sSfL --retry 3 --retry-delay 3 "$URL/$FILE" -o "$FILE"
done

cosign verify-blob "$BINARY" --signature "$BINARY".sig --certificate "$BINARY".cert --certificate-identity krel-staging@k8s-releng-prod.iam.gserviceaccount.com --certificate-oidc-issuer https://accounts.google.com

Verified OK
```

**`cosign verify` tests:**

```bash
cosign verify registry.k8s.io/kube-apiserver-amd64:v1.27.1 --certificate-identity krel-trust@k8s-releng-prod.iam.gserviceaccount.com --certificate-oidc-issuer https://accounts.google.com |
jq .

Verification for registry.k8s.io/kube-apiserver-amd64:v1.27.1 --
The following checks were performed on each of these signatures:
  - The cosign claims were validated
  - Existence of the claims in the transparency log was verified offline
  - The code-signing certificate was verified using trusted certificate authority certificates
[...]
```

```bash
curl -Ls "https://sbom.k8s.io/$(curl -Ls https://dl.k8s.io/release/stable.txt)/release" | grep "SPDXID: SPDXRef-Package-registry.k8s.io" |  grep -v sha256 | cut -d- -f3- | sed 's/-/\//' | s
ed 's/-v1/:v1/' | sort > images.txt
input=images.txt
while IFS= read -r image
do
  cosign verify "$image" --certificate-identity krel-trust@k8s-releng-prod.iam.gserviceaccount.com --certificate-oidc-issuer https://accounts.google.com | jq .
done < "$input"

Verification for registry.k8s.io/conformance-amd64:v1.27.1 --
The following checks were performed on each of these signatures:
  - The cosign claims were validated
  - Existence of the claims in the transparency log was verified offline
  - The code-signing certificate was verified using trusted certificate authority certificates
  [...]
```

File: content/en/docs/tasks/administer-cluster/verify-signed-artifacts.md [hunk 1]
```diff
@@ -11,14 +11,15 @@ weight: 420
 
 ## {{% heading "prerequisites" %}}
 
-These instructions are for Kubernetes {{< skew currentVersion >}}. If you want
+These instructions are for Kubernetes v{{< skew currentVersion >}}. If you want
```
Comment (saschagrunert) [inline content/en/docs/tasks/administer-cluster/verify-signed-artifacts.md:14]: I'm not sure we need to prepend this

File: content/en/docs/tasks/administer-cluster/verify-signed-artifacts.md [hunk 2]
```diff
@@ -41,68 +42,66 @@ for FILE in "${FILES[@]}"; do
 done
 ```
 
-Then verify the blob by using `cosign`:
+Then verify the blob by using `cosign verify-blob`:
 
 ```shell
-cosign verify-blob "$BINARY" --signature "$BINARY".sig --certificate "$BINARY".cert
+cosign verify-blob "$BINARY" --signature "$BINARY".sig --certificate "$BINARY".cert --certificate-identity krel-staging@k8s-releng-prod.iam.gserviceaccount.com --certificate-oidc-issuer https://accounts.google.com
```
Comment (saschagrunert) [inline content/en/docs/tasks/administer-cluster/verify-signed-artifacts.md:48]: Does this auto-wrap or should we wrap it manually?

Comment (tengqm) [reply content/en/docs/tasks/administer-cluster/verify-signed-artifacts.md:48]: I'd prefer a manual wrapping for this command. It is too long to fit on the screen if we don't wrap it. The command is hard to parse (in brain) if not wrapped. Horizontal scrolling works, but ... it is still hard to get an idea of the whole command.

```suggestion
cosign verify-blob "$BINARY" \
  --signature "$BINARY".sig \
  --certificate "$BINARY".cert \
  --certificate-identity krel-staging@k8s-releng-prod.iam.gserviceaccount.com \
  --certificate-oidc-issuer https://accounts.google.com
```

Comment (tengqm) [reply content/en/docs/tasks/administer-cluster/verify-signed-artifacts.md:48]: I'd prefer a manual wrapping for this command. It is too long to fit on the screen if we don't wrap it. The command is hard to parse (in brain) if not wrapped. Horizontal scrolling works, but ... it is still hard to get an idea of the whole command.

```suggestion
cosign verify-blob "$BINARY" \
  --signature "$BINARY".sig \
  --certificate "$BINARY".cert \
  --certificate-identity krel-staging@k8s-releng-prod.iam.gserviceaccount.com \
  --certificate-oidc-issuer https://accounts.google.com
```

File: content/en/docs/tasks/administer-cluster/verify-signed-artifacts.md [hunk 3]
```diff
@@ -41,68 +42,66 @@ for FILE in "${FILES[@]}"; do
...
 
-cosign v1.9.0 is required to be able to use the `--certificate` flag. Please use
-`--cert` for older versions of cosign.
-
 {{< note >}}
+Cosign 2.0 requires the `--certificate-identity` and `--certificate-oidc-issuer` options.
+
 To learn more about keyless signing, please refer to [Keyless
-Signatures](https://github.com/sigstore/cosign/blob/main/KEYLESS.md#keyless-signatures).
+Signatures](https://docs.sigstore.dev/cosign/keyless).
```
Comment (tengqm) [inline content/en/docs/tasks/administer-cluster/verify-signed-artifacts.md:55]: Don't break the links into two lines if possible.

```suggestion
[Keyless signatures](....)
```

File: content/en/docs/tasks/administer-cluster/verify-signed-artifacts.md [hunk 4]
```diff
@@ -41,68 +42,66 @@ for FILE in "${FILES[@]}"; do
...
-. {{< /note >}}
-
 ### Verifying images for all control plane components
 
-To verify all signed control plane images, please run this command:
+To verify all signed control plane images for the latest stable version (v{{< skew currentPatchVersion >}}), please run the following commands:
 
 ```shell
-curl -Ls "https://sbom.k8s.io/$(curl -Ls https://dl.k8s.io/release/stable.txt)/release" | grep "SPDXID: SPDXRef-Package-registry.k8s.io" |  grep -v sha256 | cut -d- -f3- | sed 's/-/\//' | sed 's/-v1/:v1/' > images.txt
+curl -Ls "https://sbom.k8s.io/$(curl -Ls https://dl.k8s.io/release/stable.txt)/release" | grep "SPDXID: SPDXRef-Package-registry.k8s.io" |  grep -v sha256 | cut -d- -f3- | sed 's/-/\//' | sed 's/-v1/:v1/' | sort > images.txt
```
Comment (tengqm) [inline content/en/docs/tasks/administer-cluster/verify-signed-artifacts.md:79]:
```suggestion
curl -Ls "https://sbom.k8s.io/$(curl -Ls https://dl.k8s.io/release/stable.txt)/release" \
  | grep "SPDXID: SPDXRef-Package-registry.k8s.io" \
  |  grep -v sha256 | cut -d- -f3- | sed 's/-/\//' | sed 's/-v1/:v1/' \
  | sort > images.txt
```

File: content/en/docs/tasks/administer-cluster/verify-signed-artifacts.md [hunk 5]
```diff
@@ -41,68 +42,66 @@ for FILE in "${FILES[@]}"; do
...
 input=images.txt
 while IFS= read -r image
 do
-  COSIGN_EXPERIMENTAL=1 cosign verify "$image"
+  cosign verify "$image" --certificate-identity krel-trust@k8s-releng-prod.iam.gserviceaccount.com --certificate-oidc-issuer https://accounts.google.com | jq .
```
Comment (tengqm) [inline content/en/docs/tasks/administer-cluster/verify-signed-artifacts.md:83]:
```suggestion
  cosign verify "$image" \
    --certificate-identity krel-trust@k8s-releng-prod.iam.gserviceaccount.com \
    --certificate-oidc-issuer https://accounts.google.com \
    | jq .
```

File: content/en/docs/tasks/administer-cluster/verify-signed-artifacts.md [hunk 6]
```diff
@@ -41,68 +42,66 @@ for FILE in "${FILES[@]}"; do
...
 ```
 
-Once you have verified an image, specify that image by its digest in your Pod
-manifests as per this
-example: `registry-url/image-name@sha256:45b23dee08af5e43a7fea6c4cf9c25ccf269ee113168c19722f87876677c5cb2`
-.
+Once you have verified an image, you can specify the image by its digest in your Pod
+manifests as per this example:
+
+`registry-url/image-name@sha256:45b23dee08af5e43a7fea6c4cf9c25ccf269ee113168c19722f87876677c5cb2`
```
Comment (tengqm) [inline content/en/docs/tasks/administer-cluster/verify-signed-artifacts.md:90]:
````suggestion
```console
registry-url/image-name@sha256:45b23dee08af5e43a7fea6c4cf9c25ccf269ee113168c19722f87876677c5cb2
```
```

**Comment by mrgiles** [comment]
Thank you for the feedback @tengqm. I've applied the suggested changes and re-tested the `cosign` commands, all works well.

FYI: @saschagrunert @sftim @PushkarJ

**Comment by PushkarJ** [comment]
Thank you for working on this. I really appreciate the diligence. 

Only question I have is if "jq" is really needed. If it is just to make json look better formatted then I would suggest to not add it in the docs so there is less stuff to install for the user.  Let me know if I missed anything.

**Comment by mrgiles** [comment]
Thank you for the comment, @PushkarJ. I thought about it but it's more than make it pretty, if you don't use `jq` the output is totally unintelligible. Besides, I think most Kubernetes users already have `jq` in their tool chest. You will find abundant use of it in the [Kubectl Cheat Sheet](https://kubernetes.io/docs/reference/kubectl/cheatsheet/). It may not even be necessary to list it as a prerequisite, actually, the same goes for `curl`. Let me know what you think.

**Comment by PushkarJ** [comment]
I don't have a strong preference for either having it or not in the docs but definitely please keep the prerequisites as is. 

/lgtm
